starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +82,11 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory curr = this.mongoTemplate.findById(id, Inventory.class);
+    if (curr == null) {
+      return Optional.empty();
+    }
+    this.mongoTemplate.remove(curr);
+    return Optional.of(curr);
```

Comment:
So, you can do this, but there is also a way to find and remove with MongoTemplate so you can do this all in one.

Reply:
I saw the findAndRemove but the query part scared me D:

Reply:
Do you think it would be better to figure it out with findAndRemove?